### PR TITLE
Regex Memo

### PR DESCRIPTION
**Pull request by nmaltais**
Created a simple Regex Memo feature. Includes a cheatsheet and tester inspired by https://regexr.com/

### CODE REVIEW

**Comment by CorentinTh** [comment]
Hi @nmaltais !

Thank you for your contribution in this project.
You did a really great job, I'll need you to fix few things before we can merge it on my master.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,13 @@ All notable changes to this project will be documented in this file.
...
 
-## Next
+## 1.8.1
+- [fix] Fix bugs in Regex Memo textarea
+
+## 1.8.0
+- [feat] [REGEX memo](https://it-tools.tech/regex-memo)
+
+## 1.7.0
 - [feat] [Crontab friendly generator](https://it-tools.tech/crontab-generator)
```
Comment (CorentinTh) [inline CHANGELOG.md:13]: Leave the version as next. I'll upgrade it when realising a new version

```suggestion
## Next
- [fix] Fix bugs in Regex Memo textarea
- [feat] [REGEX memo](https://it-tools.tech/regex-memo)
- [feat] [Crontab friendly generator](https://it-tools.tech/crontab-generator)
```

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "it-tools",
   "description": "",
-  "version": "1.7.0",
```
Comment (CorentinTh) [inline package.json:4]: Same here and package-lock.json

File: src/components/StyledTextarea.vue [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      rangeObj.collapse(true);
+      selectObj.removeAllRanges();
+      selectObj.addRange(rangeObj);
+    }
+  }
+}
+</script>
+
+<style lang="less">
+  .StyledTextarea {
```
Comment (CorentinTh) [inline src/components/StyledTextarea.vue:69]: Class name in kebab-case
```suggestion
  .styled-textarea {
```

File: src/components/StyledTextarea.vue [hunk 4]
```diff
@@ -0,0 +1,77 @@
+<template>
+  <div class="StyledTextarea" @input="updateInput(true)" contenteditable></div>
```
Comment (CorentinTh) [inline src/components/StyledTextarea.vue:2]: Class name in kebab-case
```suggestion
  <div class="styled-textarea" @input="updateInput(true)" contenteditable></div>
```

**Comment by CorentinTh** [comment]
In the REGEX Tester card, why didn't you use the `v-textarea` like in the Base64 string converter for example ?

And please
* remove the scroll bars
* allow the user for multiline text (when pressing enter)


image

**Comment by CorentinTh** [comment]
In the REGEX Memo card, please set a uniform padding of 16px (left, right and bottom of the card)

**Comment by nmaltais** [comment]
> In the REGEX Tester card, why didn't you use the `v-textarea` like in the Base64 string converter for example ?
> 

I might be wrong but I didn't think normal textarea or v-textarea allowed us to markup the input text. I needed to be able to highlight specific portions of the input, so I made a simple component that gives that flexibility

**Pull request closed without merging**